app/assets/javascripts/helpers/profile_photo_uploader.es6

```diff
@@ -0,0 +1,250 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+
+Diaspora.ProfilePhotoUploader = class {
+  /**
+   * Initializes a new instance of ProfilePhotoUploader
+   */
+  constructor() {
+    // get several elements we will use a few times
+    this.fileInput = document.querySelector("#file-upload");
+    this.picture = document.querySelector("#profile_photo_upload .avatar");
+    this.info = document.querySelector("#fileInfo");
+    this.cropContainer = document.querySelector(".crop-container");
+    this.spinner = document.querySelector("#file-upload-spinner");
+
+    /**
+     * Creates a button
+     * @param {string} icon - The entypo icon class.
+     * @param {function} onClick - Is called when button has been clicked.
+     */
+    this.createButton = (icon, onClick) =>
+      ($(`<button class="btn btn-default" type="button"><i class="entypo-${icon}"></i></button>`)
+        .on("click", onClick));
+
+    /**
+     * Shows a message using flash messages or alert for mobile.
+     * @param {string} type - The type of the message, e.g. "error" or "success".
+     * @param text - The text to display.
+     */
+    this.showMessage = (type, text) => { return app.flashMessages ? app.flashMessages[type](text) : alert(text); };
+
+    this.initFineUploader();
+  }
+
+  /**
+   * Initializes the fine uploader component
+   */
+  initFineUploader() {
+    this.fineUploader = new qq.FineUploaderBasic({
+      element: this.fileInput,
+      validation: {
+        allowedExtensions: ["jpg", "jpeg", "png"]
```

Comment:
Do we have any easy way of passing the list of valid extensions from the backend to the frontend? Thinking about ways to elegantly workaround #7577.

Reply:
do we have a list of accepted extensions? do we want to save them in a global variable or something?

Reply:
At the current state we have multiple places where the allowed extensions are defined.
I don't know how to bring the list of atteched types from backend to frontend (without refactoring many things).
In addition we accept gif in posts while we only accept jpg and png for profile pictures. So also making a single frontend and/or a single backend setting doesn't make sense here.
Needs more thinking in the future…